src/lib.rs

```diff
@@ -173,6 +173,38 @@ compile_error!(
 
 #[doc(hidden)]
```

Comment:
don't forget to remove this and use `#[macro_export]` instead

Reply:
Why did you like to add this `#[doc(hidden)]` attribute before? And do you mean use `#[macro_export]` for this `macros` module?

Reply:
The `#[doc(hidden)]` attribute was added to make the macros private. But if you want to make it public you have to use `#[macro_export]`. Not sure if the former was necessary back when j used it myself, but I think I had trouble hiding the macro so I just used that.

Reply:
I think it's acceptable to keep the `macros` private, it's unnecessary to make it public. I prefer to recover `#[doc(hidden)]`.

Reply:
But the whole point of `scopes!` is to make it easier to use for everyone.

Reply:
Could you mention the existence of the `scope!` macro in the token `scope` member? Those who don't take a look at the examples won't know about it. Here https://github.com/ramsayleung/rspotify/blob/master/src/oauth2.rs#L88, something like "you can use the macro `scopes!` to build it at compile time easily".

Reply:
Do you think `macro` module is necessary, or we could just put the macro in the `lib.rs` as other open source projects like `serde`: https://docs.serde.rs/src/serde/macros.rs.html#112-120

Reply:
`serde` is also inside a module, because they are in the `macros.rs` file. But if you consider it cleaner that way, then go ahead.

Reply:
I think `macros.rs` is a good idea, and I will move `json_insert` and `scope` into `macros.rs`